cargo-dist/templates/ci/github/release.yml.j2

```diff
@@ -128,6 +128,53 @@ jobs:
         with:
           name: cargo-dist-cache
           path: ~/.cargo/bin/dist
+      {{%- if "plan" in github_build_setup_jobs %}}
+      {{%- for step in github_build_setup %}}
```

Comment:
Now that this is being repeated in a number of places, how about turning it into a partial pulled in via an include statement? That way we can avoid inlining exactly the same logic in a number of places. You can see a sample `include` statement here: https://github.com/axodotdev/cargo-dist/blob/main/cargo-dist/templates/ci/github/release.yml.j2#L660

And the partials go in this directory: https://github.com/axodotdev/cargo-dist/tree/main/cargo-dist/templates/ci/github/partials